File: JAVA/LAB_JAVA/a16.java
// WAP in java to input any square matrix after that find sum of top,bottom,left and right line elements of matrix
package LAB_JAVA;
import java.util.Scanner;
class Matrix2
{
    int a[][];
    int r,c;
    Scanner s=new Scanner(System.in);
    Matrix2()
    {
        System.out.println("Enter row and column of array ");
        
        r=s.nextInt();
        c=s.nextInt();
        a=new int[r][c];
        
    }
    void input()

    {
        System.out.println("Enter "+r*c+" number :");
        for(int i=0;i<r;i++)
        {
            for(int j=0;j<c;j++)
            {
                a[i][j]=s.nextInt();
            }
        }
    }
    void output()
    {
        System.out.println("Array :");
        for(int element[]:a)
        {
            for(int element1:element)
            {
                System.out.print(element1+"\t");
            }
            System.out.println(" ");            
        }
        
    }
    void Addition()
    {
        int top=0,bottom=0,left=0,right=0;
        for(int i=0;i<r;i++)
        {
            left = left+a[i][0];
                right = right+a[i][c-1];
        }
        for(int j=0;j<c;j++)
            {
                top = top+a[0][j];
                bottom = bottom+a[r-1][j];
            }
        System.out.println("Addition of top: "+top);
        System.out.println("Addition of bottom: "+bottom);
        System.out.println("Addition of left: "+left);
        System.out.println("Addition of right: "+right);
    }
    public static void main(String[] args) 
    {
        Matrix2 mat = new Matrix2();
        mat.input();
        mat.output();
        mat.Addition();    
    }
}

File: C++/LAB_C++/a16.cpp
//  Write a program to find the sum of diagonal elements, upper triangular
// elements and the lower triangular elements.
#include <iostream>
using namespace std;

template<typename T>class  Array
{
    T *a;
    int r,c;
    public:
    Array(T *arr,int r,int c)
    {
    	a=arr;
        this->r=r;
        this->c=c;
        a = new T;
        
    }
    void input()
    {
        cout<<"Enter row/column of Square Matrix: "<<endl;
        cin>>c;
        r=c;
        cout<<"Enter "<<r*c<<" element of Square Matrix: "<<endl;
        for(int i=0;i<r;i++)
        {
            for(int j=0;j<c;j++)
            {
                cin>>*(a+i*c+j);
            }
        }
    }
    void output()
    {
        cout<<"Element of Square Matrix: "<<endl;
        for(int i=0;i<r;i++)
        {
            for(int j=0;j<c;j++)
            {
                cout<<*(a+i*c+j)<<"\t";
            }
            cout<<endl;
        }
        
    }
    private:
    T F_diagonal()
    {
        T sum=0;
        for(int j=0;j<c;j++)
        {
           sum=sum+*(a+j*c+j);
        }
        return sum;
    }
    T S_diagonal()
    {
        T sum=0;
        for(int i=0;i<r;i++)
        {
            for(int j=0;j<c;j++)
            {
                if(i+j==c-1)
                {
                    sum=sum+*(a+i*c+j);
                }
            }
        }
        return sum;
    }
    T T_element()
    {
        T sum=0;
        for(int j=0;j<c;j++)
        {
            sum=sum+*(a+j);
        }
        return sum;
    }
    T B_element()
    {
        T sum=0;
        for(int j=0;j<c;j++)
        {
            sum=sum+*(a+(r-1)*c+j);
        }
        return sum;
    }
    T L_element()
    {
        T sum=0;
        for(int i=0;i<r;i++)
        {
            sum=sum+*(a+i*c);
        }
        return sum;
    }
    T R_element()
    {
        T sum=0;
        for(int i=0;i<r;i++)
        {
            sum=sum+*(a+i*c+(c-1));
        }
        return sum;
    }
    public: 
    void display()
    {
        cout<<"Sum of Top element of matrix: "<<T_element()<<endl;
        cout<<"Sum of Bottom element of matrix: "<<B_element()<<endl;
        cout<<"Sum of Right element of matrix: "<<R_element()<<endl;
        cout<<"Sum of Left element of matrix: "<<L_element()<<endl;
        cout<<"Sum of First diagonal element of matrix: "<<F_diagonal()<<endl;
        cout<<"Sum of Second diagonal element of matrix: "<<S_diagonal()<<endl;
    }
    
    ~Array()
    {
        delete a;
    }
};
int main()
{
    int *a,row,col;
    Array <int> arr(a,row,col);
    arr.input();
    arr.output();
    arr.display();
    return 0;

}
